modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeImportsCheck.java

```diff
@@ -43,6 +43,31 @@ protected String doProcess(
 		return _fixImports(fileName, content);
 	}
 
+	private static String _fixUtilityClassesReference(
+		String className, String newClassName, String content,
+		String variableRegex) {
+
+		if (className.endsWith("Util") || !newClassName.endsWith("Util")) {
+			return content;
+		}
+
+		content = content.replaceAll(variableRegex, newClassName);
+
+		String regex = StringBundler.concat(
+			"\\n?\\t@Reference\\s+\\w+\\s+", newClassName, "\\s+[_a-z]*\\w*",
+			newClassName, ";\\n?");
+
+		Pattern pattern = Pattern.compile(regex);
+
+		Matcher matcher = pattern.matcher(content);
+
+		if (matcher.find()) {
```

Comment:
@nicolas-sss , can we use the `replaceAll` from `matcher` here? This would avoid to check the `if` first.